File: server/router/jxc/purchase.go
package jxc

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type PurchaseRouter struct{}

func (r *PurchaseRouter) InitPurchaseRouter(Router *gin.RouterGroup) {
	recordGroup := Router.Group("jxc").Use(middleware.OperationRecord())
	readOnlyGroup := Router.Group("jxc")

	// ===== 采购单 =====
	readOnlyGroup.GET("purchase/page", purchaseApi.GetPurchasePage)
	readOnlyGroup.GET("purchase/detail", purchaseApi.GetPurchaseDetail)
	recordGroup.POST("purchase", purchaseApi.CreatePurchase)
	recordGroup.PUT("purchase", purchaseApi.UpdatePurchase)
	recordGroup.DELETE("purchase", purchaseApi.DeletePurchase)
	recordGroup.PUT("purchase/audit", purchaseApi.AuditPurchase)
	recordGroup.PUT("purchase/cancel", purchaseApi.CancelPurchase)
	recordGroup.PUT("purchase/stock-in", purchaseApi.StockIn)
}

File: web/src/api/jxc/purchase.js
import service from '@/utils/request'

// 采购单
export const getPurchasePage = (params) => service.get('/jxc/purchase/page', { params })
export const getPurchaseDetail = (id) => service.get('/jxc/purchase/detail', { params: { id } })
export const createPurchase = (data) => service.post('/jxc/purchase', data)
export const updatePurchase = (data) => service.put('/jxc/purchase', data)
export const deletePurchase = (id) => service.delete('/jxc/purchase', { data: { id } })
export const auditPurchase = (id) => service.put('/jxc/purchase/audit', { id })
export const cancelPurchase = (id) => service.put('/jxc/purchase/cancel', { id })
export const stockInPurchase = (id) => service.put('/jxc/purchase/stock-in', { id })
